src/vs/base/common/fuzzyScorer.ts

```diff
@@ -460,12 +487,13 @@ function doScoreItemFuzzyMultiple(label: string, description: string | undefined
 function doScoreItemFuzzySingle(label: string, description: string | undefined, path: string | undefined, query: IPreparedQueryPiece, preferLabelMatches: boolean, fuzzy: boolean): IItemScore {
 
 	// Prefer label matches if told so or we have no description
-	if (preferLabelMatches || !description) {
+	if ((preferLabelMatches && !query.excludeQuery) || !description) {
```

Comment:
this is a bit tricky... but the idea is that exclude queries have to happen on the entire path.